README.md

```diff
@@ -1,39 +1,124 @@
+
+Hierarchy is a simple utility to merge a set of yaml or json files, based on a defined hierarchy. It is inspired by Hiera from Puppet.
 
 ## Installation
 
-Add steps for installing here.
+You can either download a binary file from GitHub releases or [compile from source](#compiling-from-source).
 
 ## Documentation
 
-Add documentaiton here, or add a link to
-the full documentation.
+The main goal of `hierarchy` is to prevent the configuration data duplication and allow for a fine-grained control over the GitOps process. The output YAML can be used to generate a `values.yml` file for GitOps tools, like [Eunomia](https://github.com/KohlsTechnology/eunomia), [Helm](https://helm.sh), and others.
+
+### Usage
+
+You can control the behavior of Hierarchy either through command line options or environment variables. The later is especially helpful if you're running it inside a container.
+
+| Command Line Flag | Environment Variable | Default | Description |
+| --- | --- | --- | --- | --- |
+| `-f, --file` | `HIERARCHY_FILE` | `./hierarchy.lst` | Path and name of the hierarchy file. |
+| `-o, --output` | `HIERARCHY_OUTPUT` | `./output.yaml` | Path and name of the output file. |
+| `-i, --filter` | `HIERARCHY_FILTER` | `(.yaml|.yml|.json)$` | Regex for file extension of files being merged |
+| `-m, --failmissing` | `HIERARCHY_FAILMISSING` | `false` | Fail if a directory in the hierarchy is missing |
+| `-d, --debug` | `HIERARCHY_DEBUG` | `false` | Print debug output |
+| `--trace` | `HIERARCHY_TRACE` | `false` | Prints a diff after processing each file. This generates A LOT of output |
+| `-V, --version` | | | Print version and build information, then exit |
+
+### Merging
+
+The `hierarchy` utility processes the YAML structure as a deep merge, with the exception of lists. Lists are completely overwritten, and therefore, it is important to keep that in mind when using them.
+
+### Hierarchy
+
+The hierarchy is defined in the file `hierarchy.lst`. This is a simple text file that lists one include folder per line and supports comments prefixed with `#`. The directories listed can be relative or absolute (try to avoid) paths. You can have directories included that are higher or lower in the structure to control their precedence. You can look at examples [here](https://github.com/KohlsTechnology/hierarchy/blob/master/testdata/).
+
+#### Example content
+```
+../defaults    #this is the first ... lowest priority
+../marketing   #this is the second
+../development #this is the third ... highest priority
+```
+
+In this case, it will load all yaml files from `../defaults`, then merge it with everything in `../marketing`, and lastly merge it with everything in `../development`. You can also use the relative path `./`, which means that it will also load variables defined in contextDir directly (same folder level as `hierarchy.lst`). You can insert `./` in any desired order in the `hierarchy.lst`, thus determining its priority.
```

Comment:
```suggestion
In this case, it will load all yaml files from `../defaults`, then merge it with everything in `../marketing`, and lastly merge it with everything in `../development`. You can also use the relative path `./`, which means that it will also load variables defined in contextDir directly (same folder level as `hierarchy.lst`). You can insert `./` in any desired order in the `hierarchy.lst`, thus determining its priority.
```
Phasing here could be improved, would like perhaps Anna or Carolyn's opinion on how to phrase better